lib/mocha.js

```diff
@@ -424,23 +455,45 @@ Mocha.prototype.run = function(fn){
   var options = this.options;
   options.files = this.files;
   var runner = new exports.Runner(suite, options.delay);
-  var reporter = new this._reporter(runner, options);
+  // For each loaded reporter constructor, create
+  // an instance and initialize it with the runner
+  var reporters = this._reporters.map(function (reporterConfig) {
+    var _reporter = reporterConfig.fn
+      , path = reporterConfig.path;
+
+    options.reporterOptions = reporterConfig.options;
```

Comment:
Kind of gross (changing the value of `this.options` per-reporter), but I did this to keep backwards compatibility with external reporters that have been using `this.options.reporterOptions`. For example, [this Slack reporter](https://github.com/ggarret/grunt-mocha-slack-reporter/blob/master/index.js#L21).

Open to alternative suggestions.